common/src/main/templates/io/netty/util/collection/KObjectHashMap.template

```diff
@@ -319,6 +322,107 @@ public class @K@ObjectHashMap<V> implements @K@ObjectMap<V>, Iterable<@K@ObjectM
         return true;
     }
 
+    public boolean containsKey(Object key) {
+        return containsKey(objectToKey(key));
+    }
+
+    @Override
+    public V get(Object key) {
+        return get(objectToKey(key));
+    }
+
+    @Override
+    public V put(@O@ key, V value) {
+        return put(objectToKey(key), value);
+    }
+
+    @Override
+    public V remove(Object key) {
+        return remove(objectToKey(key));
+    }
+
+    @Override
+    public Set<@O@> keySet() {
+        return new AbstractSet<@O@>() {
+            @Override
+            public int size() {
+                return @K@ObjectHashMap.this.size();
+            }
+
+            @Override
+            public boolean contains(Object o) {
+                return @K@ObjectHashMap.this.containsKey(o);
+            }
+
+            @Override
+            public boolean remove(Object o) {
+                return @K@ObjectHashMap.this.remove(o) != null;
+            }
+
+            @Override
+            public boolean retainAll(Collection<?> retainedKeys) {
+                boolean changed = false;
+                for(Iterator<PEntry<V>> iter = @K@ObjectHashMap.this.entries().iterator(); iter.hasNext(); ) {
+                    PEntry<V> entry = iter.next();
+                    if (!retainedKeys.contains(entry.key())) {
+                        changed = true;
+                        iter.remove();
+                    }
+                }
+                return changed;
+            }
+
+            @Override
+            public void clear() {
+                @K@ObjectHashMap.this.clear();
+            }
+
+            @Override
+            public Iterator<@O@> iterator() {
+                return new Iterator<@O@>() {
+                    private Iterator<PEntry<V>> iter = @K@ObjectHashMap.this.entries().iterator();
+
+                    @Override
+                    public boolean hasNext() {
+                        return iter.hasNext();
+                    }
+
+                    @Override
+                    public @O@ next() {
+                        return iter.next().key();
+                    }
+
+                    @Override
+                    public void remove() {
+                        iter.remove();
+                    }
+                };
+            }
+        };
+    }
+
+    @Override
+    public Set<Map.Entry<@O@, V>> entrySet() {
+        return new AbstractSet<Map.Entry<@O@, V>>() {
+            @Override
+            public Iterator<Map.Entry<@O@, V>> iterator() {
+                return new IteratorWrapper<V>(new IteratorImpl());
+            }
+
+            @Override
+            public int size() {
+                return @K@ObjectHashMap.this.size();
+            }
+        };
+    }
+
+    private @k@ objectToKey(Object key) {
+        if (!(key instanceof @O@)) {
```

Comment:
that shouldn't be necessary. if it's not of correct type the cast will fail anyway ...